WatchDogIntelliJPlugin/WatchDog/src/nl/tudelft/watchdog/intellij/WatchDog.java

```diff
@@ -70,20 +90,31 @@ public void projectOpened() {
 
     public void projectClosed() {
         // called when project is being closed
-        InitializationManager intervalInitializationManager = InitializationManager
-                .getInstance();
+        if(!WatchDogUtils.isWatchDogActive(project)) {
+            return;
+        }
+        InitializationManager intervalInitializationManager = InitializationManager.getInstance(project.getName());
         intervalInitializationManager.getEventManager().update(new WatchDogEvent(this, WatchDogEvent.EventType.END_IDE));
         intervalInitializationManager.getIntervalManager().closeAllIntervals();
         intervalInitializationManager.getTransferManager().sendIntervalsImmediately();
-        intervalInitializationManager.shutdown();
+        intervalInitializationManager.shutdown(project.getName());
```

Comment:
Don't we have this exact same followup of steps already somewhere? Look at TimeSynchronityChecker and factor our commonalities